City-of-Helsinki/berth-reservations
# VEN-522 | Remove OverwriteStorage

## Description :sparkles:
* Replace `OverwriteStorage` with regular `FileSystemStorage`
* Add tests to ensure that the files are being removed
* Update the `resources.mutations.0001_initial` to remove the `OverwriteStorage`
* Bug fix on `get_map_resource_media_folder` that was using the `HarborMap/WinterStorageAreaMap` id to get the dir instead of the parent `Harbor/WSA` id.

## Issues :bug:
**Closes :no_good_woman:**
**[VEN-522](https://helsinkisolutionoffice.atlassian.net/browse/VEN-522)**: Remove OverwriteStorage from image_file fields

## Testing :alembic:
### Automated tests :gear:️
```shell
$ pytest resources/tests/test_resources_models.py
```

### Manual testing :construction_worker_man:
(Same instructions apply for `WinterStorageArea`)

**Replace image:**
1. Create a `Harbor` with an image
2. Check the `{MEDIA_ROOT}/harbors/{harbor_id}` directory, it should only have the image uploaded.
3. Update the `harbor` and add a new image
4. Check the harbor directory, it should _only_ have the new image

**Remove image:**
1. [Idem](https://en.wikipedia.org/wiki/Idem)
2. Idem
3. Update the `harbor` by removing the image
4. Check the harbor directory, it should be empty

**Several files:**
1. Create a `Harbor` with an image and one (or more) map files
2. Check the `{MEDIA_ROOT}/harbors/{harbor_id}` directory, it should have all the files

**Directory removed:**
1. Idem
2. Idem
3. Delete the `harbor`
4. Check the `{MEDIA_ROOT}/harbors/` directory, the `harbor.id` directory should be gone
